### PR TITLE
feature(groups): group avatars now use serve-file handler

### PR DESCRIPTION
**Pull request by hypeJunction**
Group avatars are now served with elgg_get_inline_url()

### CODE REVIEW

**Comment by hypeJunction** [comment]
@jdalsem @jeabakker Could you check this works with all of your custom handlers? I don't expect any problems, but would be nice to double check.

**Comment by mrclay** [comment]
No hooks/resource view to worry about here :+1:

File: mod/groups/icon.php [hunk 1]
```diff
@@ -1,57 +1,26 @@
...
  *
  * @package ElggGroups
  */
+elgg_deprecated_notice('icon.php has been depreated and should not be included. Use elgg_get_inline_url() instead.', '2.1');
 
-if (!isset($page)) {
-	die('This file cannot be called directly.');
-}
+$guid = get_input('group_guid');
+$size = get_input('size') ? : 'medium';
```
Comment (jeabakker) [inline mod/groups/icon.php]: why not 
```php
$size = get_input('size', 'medium');
```

**Comment by jeabakker** [comment]
I think this looks good

File: mod/groups/icon.php (hunk 1 above)
Comment (hypeJunction) [inline mod/groups/icon.php]: `?size=` would give an empty string and not default to `medium` AFAIK

Comment (jeabakker) [inline mod/groups/icon.php]: true (i think), but if i for some stupid reason have a custom size ```0``` i could never get it ;)
what would be the best way to solve this?

Comment (hypeJunction) [inline mod/groups/icon.php]: Lol. Just don't :)

Comment (hypeJunction) [inline mod/groups/icon.php]:
```php
$size = get_input('size');
if (!isset($size) || $size === '') {
   $size = 'medium';
}
```

Comment (jeabakker) [inline mod/groups/icon.php]: I love it!

**Comment by hypeJunction** [comment]
@jeabakker Updated size variable handling
Updated deprecation messages to 2.2

**Comment by mrclay** [comment]
Should mention deprecations in docs/guides/upgrading (until there's a better place).

**Comment by hypeJunction** [comment]
@mrclay Adding a second `deprecate` commit that updates the docs

**Pull request merged**